discretize/BaseMesh.py

```diff
@@ -1,41 +1,57 @@
 import numpy as np
+import properties
+import os
+import json
 from discretize import utils
 
 
-class BaseMesh(object):
-    """BaseMesh does all the counting you don't want to do.
+class BaseMesh(properties.HasProperties):
+    """
+    BaseMesh does all the counting you don't want to do.
     BaseMesh should be inherited by meshes with a regular structure.
-
-    :param numpy.array n: (or list) number of cells in each direction (dim, )
-    :param numpy.array x0: (or list) Origin of the mesh (dim, )
-
     """
 
+    # Properties
+    n = properties.Array(
+        "number of cells in each direction (dim, )",
+        dtype=int,
+        required=True,
```

Comment:
Do we want `min_length` or `max_length` here?

http://propertiespy.readthedocs.io/en/latest/content/container.html#properties.List

Reply:
Actually, that is only for lists. Perhaps we could change properties to have a `set` of tuples for defining the shape?

`shape={(1,), (2,), (3,)}`??

Reply:
not sure if we need something that complex? this is just 3 numbers (at most), nCx, nCy, nCz

Reply:
That would be handy